### DISCUSSION TITLE
Obsługa sterownika ST-505

### DISCUSSION THREAD

**Discussion by bazylewicz** [mariusz-ostoja-swierczynski/tech-controllers#92] [General]
Dysponuję sterownikiem Defro ST-505, więc mogę pomóc przygotować integrację pod niego (odczytać / podsłuchać komunikację itp).
Jeżeli jest taka chęć to proszę autorów o pomoc w tej sprawie,
Rafal

**Comment by daroga0002** [comment]
Ta integracja bazuje na uzyciu chmury emodul.pl, jezeli wspomniany sterownik integruje sie z tą usługa Tech to przynajmniej w ograniczonym zakresie to powinno działac.

**Comment by bazylewicz** [comment]
Sterownik obsługuje rekuperator Defro. Dodałem integrację do HA, wpisałem dane do logowania emodul, połączyło pomyślnie ale nie dodało żadnych zmiennych..

**Comment by skapron** [comment]
Witam, moduł ST-505 nie jest sterownikiem tylko bramką.
W moim przypadku jest spięty przez RS ze sterownikiem K1Pv4 i jedyna wersja obsługująca te sterowniki to wersja "dev" która obsługuje "tiles", niestety obsługa płytek nie została zaimplementowana w najnowszych wersjach. Jest możliwość adaptacji kodu z wersji "dev"?

Edit:
https://github.com/mariusz-ostoja-swierczynski/tech-controllers/tree/dev
Ta wersja wspiera st-505, ale jest mocno przedawniona.

**Reply by daroga0002** [reply]
Hmm, ten branch jest naprawde mocno do tylu. 

Prosba wrzućcie co widać pod https://emodul.pl/ dla tego urzadzenia oraz odpowiedz z API, mozna wyciagnac jak w instrukcji:
https://github.com/mariusz-ostoja-swierczynski/tech-controllers/blob/main/CONTRIBUTING.md#getting-json-data-from-emodul


dane wrazliwe mozna zastapic gwiazdkami, wtedy bedzie wiadome czego brakuje w obecnej integracji

**Comment by bazylewicz** [comment]
W załączeniu:

[kod.txt](https://github.com/user-attachments/files/15863986/kod.txt)
![Bez tytułu](https://github.com/mariusz-ostoja-swierczynski/tech-controllers/assets/75644854/c6733359-16fa-4609-bc4f-afc882f18330)

**Comment by skapron** [comment]
Ogólnie wersja dev działa na dzień dzisiejszy, ale to się zmieni:

`Rejestrator: homeassistant.components.sensor
Źródło: helpers/deprecation.py:206
integracja: Sensor (dokumentacja, Problemy)
Pierwsze zdarzenie: 9 czerwca 2024 03:10:40 (168686 zdarzenia)
Ostatnio zalogowany: 11:59:36

DEVICE_CLASS_TEMPERATURE was used from tech, this is a deprecated constant which will be removed in HA Core 2025.1. Use SensorDeviceClass.TEMPERATURE instead, please report it to the author of the 'tech' custom integration
DEVICE_CLASS_BATTERY was used from tech, this is a deprecated constant which will be removed in HA Core 2025.1. Use SensorDeviceClass.BATTERY instead, please report it to the author of the 'tech' custom integration`

**Comment by skapron** [comment]
[emodul.json](https://github.com/user-attachments/files/15900373/emodul.json)
W załączeniu odpowiedź z emodul.
Sterownik to Defro K1Pv4, komunikacja poprzez ST-505.
Dodam jeszcze, że w wersji dev wartości mogą być jedynie odczytywane.

**Reply by daroga0002** [reply]
patrzac na odpowiedzi to tutaj Tech daje w API tylko odpowiedzi dotyczace `tiles` i nic wiecej wartosciowego. 

W swoim API tez nie dokumentuja tego https://api-documentation.emodul.eu/ wiec ciezko jest tutaj zrozumiec skad sie biora te dodatkowe pola.

**Reply by daroga0002** [reply]
dobra wyglada ze tutaj do tych "ruchomych" elementow wykorzystuja API menu 

<img width="1478" alt="image" src="https://github.com/mariusz-ostoja-swierczynski/tech-controllers/assets/20227477/0e880d64-d624-48bc-af4a-6aec9dcd03b6">



w przegladarce powinny byc jeszcze takie pliki:

'MU', 'MI', 'MS', 'MP'
ktore tez powinny definiowac obsluge poszczegolnych menu

**Comment by bazylewicz** [comment]
Na wersji DEV działają temperatury.
MI i MU w załączeniu,

[MI.txt](https://github.com/user-attachments/files/15902615/MI.txt)
[MU.txt](https://github.com/user-attachments/files/15902629/MU.txt)

**Reply by skapron** [reply]
[MP.json](https://github.com/user-attachments/files/15923925/MP.json)
[MI.json](https://github.com/user-attachments/files/15923926/MI.json)
[MS.json](https://github.com/user-attachments/files/15923927/MS.json)
[MU.json](https://github.com/user-attachments/files/15923928/MU.json)
Cały komplet wyżej.

**Comment by bazylewicz** [comment]
Cześć, czy aktualna wersja programu czyta te zmienne, do których udostępniłem MI i MU?

**Reply by daroga0002** [reply]
nie, jest https://github.com/mariusz-ostoja-swierczynski/tech-controllers/issues/141 ale ciezko powiedziec kiedy zostanie to zrealizowane (byc moze nigdy) bo wymaga to dosc duzo dewelopmentu nowego

**Reply by bazylewicz** [reply]
Czyli po aktualizacji do HA 2025.01 integracja przestanie działać..? A czy można "ręcznie" wyciągać jakieś pojedyncze dane z tego ich API wpisując konfigurację w yaml?

**Reply by daroga0002** [reply]
integracja dziala i bedzie dzialac. To co tutaj poruszamy to sa nowe integracje ktorych obecnie nie ma wogole w obsludze i trzeba poswiecic duzo pracy zeby je dodac.

**Comment by skapron** [comment]
Niestety nie działa ze sterownikami, które opisujemy tutaj.
Odczytuje jedynie cześć informacji, lecz brak najważniejszych tj: temperatura CO, CWU.
Jedynie wersja o której wspominam wyżej działała lecz po ostatniej aktualizacji HA, przestała funkcjonować.

**Reply by bazylewicz** [reply]
Niestety.. A czy jest możliwość wyciągnięcia pojedynczych zmiennych przez API umieszczając odpowiedni wpis w yaml?

**Reply by nikersik1** [reply]
Wersja "widgets" działa
https://github.com/mariusz-ostoja-swierczynski/tech-controllers/tree/feat_tiles_widgets

**Reply by skapron** [reply]
Potwierdzam, wartości temperatur już pojawiają się prawidłowo, za wyjątkiem temperatury zaworów.
W HA można odczytać jedynie % otwarcia, natomiast w emodul mamy odczyty:
![Screenshot_2025-01-26-23-14-09-44_4606ec11ac6997a4db01d54fb2d88518](https://github.com/user-attachments/assets/9730663f-8d6f-4e18-9c47-587812241359)

**Reply by nikersik1** [reply]
![image](https://github.com/user-attachments/assets/0ae7a440-c7e2-4d40-8628-4f3fbd45fa82)

**Reply by nikersik1** [reply]
Jest w atrybutach zaworu

**Reply by skapron** [reply]
Faktycznie, nie dopatrzyłem.
Poniżej templatka jak wyciągnąć to jako dodatkową encję.
```
# Temperature from entity
sensor:
  - platform: template
    sensors:
      kotlownia_temperatura_zaworu:
        friendly_name: "Temperatura zaworu CO"
        unit_of_measurement: '°C'
        device_class: temperature
        value_template: "{{state_attr('sensor.kotlownia_zawor_wbudowany_0','currentTemp') }}"
```

**Comment by bazylewicz** [comment]
Wersja  [feat_tiles_widgets](https://github.com/mariusz-ostoja-swierczynski/tech-controllers/tree/feat_tiles_widgets)  działa poprawnie, ale brakuje wartości dwóch zmiennych:  przepływ nawiewu i wywiewu.  Analizując dane odbierane po API znalazłem ich numery, po których są identyfikowane.. Czy mogę je jakoś ręcznie teraz dodać np. do pliku const.py żeby pojawiły się w HA?

**Reply by daroga0002** [reply]
to raczej trzeba obsluzyc pewnie w kodzie, mozesz tez wkleic tutaj pelne sekcje outputow z API i wskaz ktore to.

**Reply by anarion80** [reply]
Wszystko się niestety obecnie zasadza na tile type:

```python
# tile type
TYPE_TEMPERATURE = 1
TYPE_FIRE_SENSOR = 2
TYPE_WIDGET = 6
TYPE_RELAY = 11
TYPE_ADDITIONAL_PUMP = 21
TYPE_FAN = 22
TYPE_VALVE = 23
TYPE_MIXING_VALVE = 24
TYPE_FUEL_SUPPLY = 31
TYPE_TEXT = 40
TYPE_SW_VERSION = 50
TYPE_OPEN_THERM = 252
```

A nie ma jednej jasnej definicji, jak się to mapuje do widgetów w emodul. Znalazłem conajmniej jeden przypadek, gdzie typy się pokrywały, a widgety były różne i wymagały różnej obsługi. Stąd temat trochę umarł.

**Comment by bazylewicz** [comment]
Przepływ ma wartość 204 (m3/h) i numer txtId 6131, drugi przepływ też 204 m3/h i txtId 5994:

`{"tiles":[{"id":5761,"parentId":0,"type":1,"menuId":0,"orderId":1,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":4460,"value":229,"batteryLevel":null,"signalStrength":null}},{"id":6039,"parentId":0,"type":1,"menuId":0,"orderId":4,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":4461,"value":226,"batteryLevel":null,"signalStrength":null}},{"id":6073,"parentId":0,"type":41,"menuId":0,"orderId":15,"visibility":true,"params":{"description":"Date","year":2025,"month":4,"day":14,"dayId":1,"hours":12,"minutes":11}},{"id":6153,"parentId":0,"type":1,"menuId":0,"orderId":5,"visibility":true,"params":{"description":"Temperature sensor","workingStatus":true,"txtId":6490,"value":229,"batteryLevel":null,"signalStrength":null}},{"id":6341,"parentId":0,"type":6,"menuId":0,"orderId":6,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":102,"widget1":{"txtId":0,"value":0,"unit":-1,"type":0,"params":[0,0,0,0,0]},"widget2":{"txtId":6131,"value":204,"unit":0,"type":1,"params":[0,0,0,0,0]}}},{"id":6342,"parentId":0,"type":6,"menuId":0,"orderId":7,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":102,"widget1":{"txtId":0,"value":0,"unit":-1,"type":0,"params":[0,0,0,0,0]},"widget2":{"txtId":5994,"value":204,"unit":0,"type":1,"params":[0,0,0,0,0]}}}],"zones":{"elements":[],"globalSchedules":{"elements":[]}},"menu":[],"parents":[],"lastUpdate":"2025-04-14T13:11:46.568608+02:00","moduleStatus":"active","controllerStatus":"active","init":false}`

Edit:  z tego co widzę niektóre wartości są dostępne od razu w tiles -> ID -> params -> value,  a te których mi brakuje są w  tiles -> ID -> params -> widget2 -> value.

**Comment by anarion80** [comment]
Jak widać jest za mało danych:

```json
{"id":6341,"parentId":0,"type":6,"menuId":0,"orderId":6,"visibility":true,"params":{"description":"Universal status with widgets","statusId":1,"iconId":102,"widget1":{"txtId":0,"value":0,"unit":-1,"type":0,"params":[0,0,0,0,0]},"widget2":{"txtId":6131,"value":204,"unit":0,"type":1,"params":[0,0,0,0,0]}}},
```

Mamy tylko value i txtid, unit jest 0 (czyli nie znamy jednostki), type jest "1" czyli na podstawie wcześniejszych przykładów użytkowników jest to pompa CWU, co jak widać nie jest prawdą.

Dodałem jeden commit na branch `feat_tiles_widgets` gdzie zakładam (bez żadnych testów) powinien się ten przepływ teraz pojawić jako temperatura.

**Reply by bazylewicz** [reply]
Działa, dziękuję  :-)

**Comment by Misiek1977** [comment]
Szybkie Pytanie. Też posiadam bramkę ST-505 podpiętą pod pompę Galmet (sterownik ST-53.4 G). Udało mi się odpalić odczytywanie danych. Czy jest szansa na sterowanie pompą (może coś źle robię). Pozdrawiam i zgóry dziękuje.

**Comment by bigjeans** [comment]
Cześć, również używam ST-505 ale mam dziwny odczyt dla pompy, pokazuje 0-10, ale w stopniach Celsjusza lub innych jednostkach ?

![2025-06-24_09h26_05](https://github.com/user-attachments/assets/cac728f3-9974-4ea2-b2d1-61388e349068)

**Reply by anarion80** [reply]
@bigjeans Podeślij plik JSON zgodnie z instrukcją: https://github.com/mariusz-ostoja-swierczynski/tech-controllers/blob/main/CONTRIBUTING.md#getting-json-data-from-emodul

**Reply by bigjeans** [reply]
Nie jestem pewien czy dobrze to zrobiłem, ale coś mam ;)
[emodul.json](https://github.com/user-attachments/files/20881647/emodul.json)

![2025-06-24_12h18_11](https://github.com/user-attachments/assets/42da0d17-f15f-4ceb-8be7-e997d73a37a6)
